cli/cmd/resource_groups.go

```diff
@@ -0,0 +1,291 @@
+					Groups []resourceGroup `json:"resource_groups"`
+				}{Groups: groups}
+				return cli.OutputJSON(jsonOut)
+			}
+
+			rows := [][]string{}
+			for _, g := range groups {
+				rows = append(rows, []string{g.Id, g.ResType, g.Name, g.State})
+			}
+
+			cli.OutputHuman(renderSimpleTable([]string{"RESOURCE ID", "TYPE", "NAME", "STATE"}, rows))
+			return nil
+		},
+	}
+	// show command is used to retrieve a lacework resource group by resource id
+	resourceGroupsShowCommand = &cobra.Command{
+		Use:   "show",
+		Short: "get resource group by id",
+		Long:  "Get a single resource group by it's Resource ID.",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(_ *cobra.Command, args []string) error {
+			var response api.ResourceGroupResponse
+			err := cli.LwApi.V2.ResourceGroups.Get(args[0], &response)
+			if err != nil {
+				return errors.Wrap(err, "unable to get resource group")
+			}
+
+			group := resourceGroup{
+				Id:      response.Data.ResourceGuid,
+				ResType: response.Data.Type,
+				Name:    response.Data.Name,
+				State:   response.Data.Status(),
+				Props:   response.Data.Props,
+			}
+
+			if cli.JSONOutput() {
+				jsonOut := struct {
+					Group resourceGroup `json:"resource_group"`
+				}{Group: group}
+				return cli.OutputJSON(jsonOut)
+			}
+
+			groupCommon := [][]string{}
+			groupCommon = append(groupCommon, []string{group.Id, group.ResType, group.Name, group.State})
+
+			cli.OutputHuman(renderSimpleTable([]string{"RESOURCE ID", "TYPE", "NAME", "STATE"}, groupCommon))
+			cli.OutputHuman("\n")
+			cli.OutputHuman(buildResourceGroupPropsTable(group))
+
+			return nil
+		},
+	}
+
+	// delete command is used to remove a lacework resource group by resource id
+	resourceGroupsDeleteCommand = &cobra.Command{
+		Use:   "delete",
+		Short: "delete a resource group",
+		Long:  "Delete a single resource group by it's Resource ID.",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(_ *cobra.Command, args []string) error {
+			err := cli.LwApi.V2.ResourceGroups.Delete(args[0])
+			if err != nil {
+				return errors.Wrap(err, "unable to delete resource group")
+			}
+			return nil
+		},
+	}
+
+	// create command is used to create a new lacework resource group
+	resourceGroupsCreateCommand = &cobra.Command{
+		Use:   "create",
+		Short: "create a new resource group",
+		Long:  "Creates a new single resource group.",
+		RunE: func(_ *cobra.Command, args []string) error {
+			if !cli.InteractiveMode() {
+				return errors.New("interactive mode is disabled")
+			}
+
+			err := promptCreateResourceGroup()
+			if err != nil {
+				return errors.Wrap(err, "unable to create resource group")
+			}
+
+			cli.OutputHuman("The resource group was created.\n")
+			return nil
+		},
+	}
+)
+
+func promptCreateResourceGroup() error {
+	var (
+		group  = ""
+		prompt = &survey.Select{
+			Message: "Choose a resource group type to create: ",
+			Options: []string{
+				"Aws",
+				"Azure",
+				"Container",
+				"Gcp",
+				"LwAccount",
+				"Machine",
```

Comment:
Could we capitalize all f these types? Just so that they match with our "schema" from the backend
```suggestion
				"MACHINE",
```